BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -192,10 +192,15 @@ extension AttributedLabel {
             let textContainer = NSTextContainer()
 
             textContainer.lineFragmentPadding = 0
-            textContainer.lineBreakMode = lineBreakMode
             textContainer.maximumNumberOfLines = numberOfLines
             textContainer.size = textRect(forBounds: bounds, limitedToNumberOfLines: numberOfLines).size
 
+            /// This can only be set if we have one line because NSTextContainer assumes the number of lines is 1 if
+            /// anything other than wrapping by word or character. This should still work correctly usually.
```

Comment:
`Label` does this:
```swift
    private var effectiveLineBreakMode: NSLineBreakMode {
        // These line break modes don't work when numberOfLines is 1, and they break line height adjustments.
        // Normalize them to clipping mode instead.
        if numberOfLines == 1 && (lineBreakMode == .byCharWrapping || lineBreakMode == .byWordWrapping) {
            return .byClipping
        }
        return lineBreakMode
    }
```
Can we do that here? Maybe extract it to an extension on `NSLineBreakMode`?

Reply:
- [ ] Extract `effectiveLineBreakMode`